streaming/sender.c

```diff
@@ -507,14 +489,13 @@ if(!s->rrdpush_compression)
 
     http[received] = '\0';
     debug(D_STREAM, "Response to sender from far end: %s", http);
-    int32_t version = (int32_t)parse_stream_version(host, http);
+    int32_t version = parse_stream_version(host, http);
```

Comment:
@ktsaou the unique report from Coverity is this:

```sh
*** CID 357382:  Insecure data handling  (TAINTED_SCALAR)
/streaming/sender.c: 492 in rrdpush_sender_thread_connect_to_parent()
486             rrdpush_sender_thread_close_socket(host);
487             return 0;
488         }
489     
490         http[received] = '\0';
491         debug(D_STREAM, "Response to sender from far end: %s", http);
>>>     CID 357382:  Insecure data handling  (TAINTED_SCALAR)
>>>     Passing tainted expression "http" to "parse_stream_version", which uses it as a loop boundary.
492         int32_t version = parse_stream_version(host, http);
493         if(version == -1) {
494             worker_is_busy(WORKER_SENDER_JOB_DISCONNECT_BAD_HANDSHAKE);
495             error("STREAM %s [send to %s]: server is not replying properly (is it a netdata?).", rrdhost_hostname(host), s->connected_to);
496             rrdpush_sender_thread_close_socket(host);
497             //catch other reject reasons and force to check other destinations
```

, but we are setting the last character as `NULL` few lines before we call the function, and inside function we do not modify it.